### ISSUE TITLE
Ubuntu installation failure

### ISSUE THREAD

**Issue by howarddavis** [hestiacp/hestiacp#159]
### Operating System (OS/VERSION):

Ubuntu 18.04 LTS

### HestiaCP Version:

master (current)

### Installed Software (what you got with the installer):

default (all?)

### Steps to Reproduce:

Simply run the downloaded hst-install.sh on Ubuntu 18.04 LTS after removing the preinstalled ufw firewall. Press y to install, enter admin email address, enter FQDN hostname, then it will attempt to install, followed by an error on line 800 of [hst-install-ubuntu.sh](https://github.com/hestiacp/hestiacp/blob/master/install/hst-install-ubuntu.sh) (ntpdate: command not found), then several more errors (output pasted below in Other Notes).

### Related Issues/Forum Threads:

https://github.com/hestiacp/hestiacp/issues/117 (possibly related)

### Other Notes:

**Output:**
```
root@cms:~# bash hst-install.sh
Please wait a few seconds, we update your repository before we start the installation process...


  _   _           _   _        ____ ____  
 | | | | ___  ___| |_(_) __ _ / ___|  _ \ 
 | |_| |/ _ \/ __| __| |/ _` | |   | |_) |
 |  _  |  __/\__ \ |_| | (_| | |___|  __/ 
 |_| |_|\___||___/\__|_|\__,_|\____|_|    

                      Hestia Control Panel



The following software will be installed on your system:
   - Nginx Web Server
   - Apache Web Server (as backend)
   - Bind DNS Server
   - Exim Mail Server
   - Dovecot POP3/IMAP Server
   - MariaDB Database Server
   - Vsftpd FTP Server
   - Iptables Firewall + Fail2Ban


Would you like to continue [y/n]: y
Please enter admin email address: **REDACTED**
Please enter FQDN hostname [cms.**REDACTED**]: cms.**REDACTED**

Installation backup directory: /root/hst_install_backups/220220191332
Installation Log File: /root/hst_install_backups/hst_install-220220191332.log

Upgrade System using apt-get...\
Install third party repository keys... 
Install HestiaCP and all required packages, the process will take around 10-15 minutes.../
hst-install-ubuntu.sh: line 800: ntpdate: command not found
sed: can't read /etc/rssh.conf: No such file or directory
sed: can't read /etc/rssh.conf: No such file or directory
sed: can't read /etc/rssh.conf: No such file or directory
chmod: cannot access '/usr/bin/rssh': No such file or directory
cp: cannot stat '/usr/local/hestia/install/ubuntu/18.04/sudo/admin': No such file or directory
chmod: cannot access '/etc/sudoers.d/admin': No such file or directory
cp: cannot stat '/usr/local/hestia/install/ubuntu/18.04/logrotate/hestia': No such file or directory
cp: cannot stat '/usr/local/hestia/install/ubuntu/18.04/packages': No such file or directory
cp: cannot stat '/usr/local/hestia/install/ubuntu/18.04/templates': No such file or directory
cp: cannot stat '/usr/local/hestia/data/templates/web/skel/public_html/index.html': No such file or directory
sed: can't read /var/www/index.html: No such file or directory
cp: cannot stat '/usr/local/hestia/install/ubuntu/18.04/firewall': No such file or directory
hst-install-ubuntu.sh: line 952: /usr/local/hestia/bin/v-generate-ssl-cert: No such file or directory
sed: -e expression #1, char 3: unexpected `,'
sed: -e expression #1, char 1: unknown command: `,'
cp: cannot stat '/usr/local/hestia/install/ubuntu/18.04/nginx/nginx.conf': No such file or directory
cp: cannot stat '/usr/local/hestia/install/ubuntu/18.04/nginx/status.conf': No such file or directory
cp: cannot stat '/usr/local/hestia/install/ubuntu/18.04/nginx/phpmyadmin.inc': No such file or directory
cp: cannot stat '/usr/local/hestia/install/ubuntu/18.04/nginx/phppgadmin.inc': No such file or directory
cp: cannot stat '/usr/local/hestia/install/ubuntu/18.04/nginx/webmail.inc': No such file or directory
cp: cannot stat '/usr/local/hestia/install/ubuntu/18.04/logrotate/nginx': No such file or directory
hst-install-ubuntu.sh: line 986: /etc/nginx/conf.d/hestia.conf: No such file or directory
Failed to start nginx.service: Unit nginx.service not found.
Error: nginx start failed
```
It appears that ntpdate is not installed by default on Ubuntu 18.04 LTS, perhaps we need to include its installation in the bash file IF it is not already installed.
**[line 796-800 in hst-install-ubuntu.sh](https://github.com/hestiacp/hestiacp/blob/master/install/hst-install-ubuntu.sh#L796):**
```
# Configuring NTP
echo '#!/bin/sh' > /etc/cron.daily/ntpdate
echo "$(which ntpdate) -s pool.ntp.org" >> /etc/cron.daily/ntpdate
chmod 755 /etc/cron.daily/ntpdate
ntpdate -s pool.ntp.org
```
It appears that rssh isn't installed by default on Ubuntu 18.04 LTS, perhaps we need to include its installation in the bash file IF it is not already installed.
**[line 802-809 in hst-install-ubuntu.sh](https://github.com/hestiacp/hestiacp/blob/master/install/hst-install-ubuntu.sh#L802):**
```
# Setup rssh
if [ -z "$(grep /usr/bin/rssh /etc/shells)" ]; then
    echo /usr/bin/rssh >> /etc/shells
fi
sed -i 's/#allowscp/allowscp/' /etc/rssh.conf
sed -i 's/#allowsftp/allowsftp/' /etc/rssh.conf
sed -i 's/#allowrsync/allowrsync/' /etc/rssh.conf
chmod 755 /usr/bin/rssh
```

**Comment by ScIT-Raphael** [comment]
Hi, thanks for your answer. I just checked the installer and fixed a bug with the latest commit: https://github.com/hestiacp/hestiacp/commit/8ce248c007a7db770704e27f01724a3b7bd2b2ef

Would you like to check it again and let me know, if all works?

**Comment by howarddavis** [comment]
Bravo!

Only one warning in the output, but everything appeared to install properly. The control panel is up and running.

```
...
Upgrade System using apt-get...|
Install third party repository keys... 
Install HestiaCP and all required packages, the process will take around 10-15 minutes...-
2019-02-22 15:00:26 0 [Warning] option 'table_open_cache': unsigned value 4 adjusted to 10
Upgrade phpMyAdmin to v4.8.5...

Congratulations, you have just successfully installed Hestia Control Panel
...
```

**Comment by ScIT-Raphael** [comment]
Hi @howarddavis 
I'm happy that it works now, I've added already a second commit to do a better handling with the fpm conditions: https://github.com/hestiacp/hestiacp/commit/be69edcd5d037d727b7b9dc764211ea010d7efb1

Also for the note you've sent me, I adjusted the value to 10 as requested by mariadb with the following commit: https://github.com/hestiacp/hestiacp/commit/7c5708abff1fb130b3c8f1301a558b3f41525932

Because all issues are fixed now, I will close this issue. Please feel free to reopen it or create a new one, if you find any additional issue or also if you've a feature request.

**Comment by KamuiX** [comment]
Tried today to install hestiacp on Ubuntu 18.04 the above problem poped up again.

> Updating currently installed packages, please wait...|
> Now installing Hestia Control Panel and all required dependencies.
> NOTE: This process may take 10 to 15 minutes to complete, please wait...                                                         /
> (*) Configuring system settings...
> (*) Configuring Hestia Control Panel...
> cp: cannot stat '/usr/local/hestia/install/deb/sudo/admin': No such file or directory
> chmod: cannot access '/etc/sudoers.d/admin': No such file or directory
> cp: cannot stat '/usr/local/hestia/install/deb/logrotate/hestia': No such file or directory
> cp: cannot stat '/usr/local/hestia/install/deb/packages': No such file or directory
> sed: can't read /usr/local/hestia/data/packages/*.pkg: No such file or directory
> cp: cannot stat '/usr/local/hestia/install/deb/templates': No such file or directory
> cp: cannot stat '/usr/local/hestia/install/deb/templates/web/unassigned/index.html': No such file or directory
> cp: cannot stat '/usr/local/hestia/install/deb/templates/web/skel/document_errors/*': No such file or directory
> cp: cannot stat '/usr/local/hestia/install/deb/firewall': No such file or directory
> (*) Generating default self-signed SSL certificate...
> hst-install-ubuntu.sh: line 1069: /usr/local/hestia/bin/v-generate-ssl-cert: No such file or directory
> (*) Adding SSL certificate to Hestia Control Panel...
> sed: -e expression #1, char 3: unexpected `,'
> sed: -e expression #1, char 1: unknown command: `,'
> cp: cannot stat '/usr/local/hestia/install/deb/ssl/dhparam.pem': No such file or directory
> (*) Configuring NGINX...
> cp: cannot stat '/usr/local/hestia/install/deb/nginx/nginx.conf': No such file or directory
> cp: cannot stat '/usr/local/hestia/install/deb/nginx/status.conf': No such file or directory
> cp: cannot stat '/usr/local/hestia/install/deb/nginx/phpmyadmin.inc': No such file or directory
> cp: cannot stat '/usr/local/hestia/install/deb/nginx/phppgadmin.inc': No such file or directory
> cp: cannot stat '/usr/local/hestia/install/deb/logrotate/nginx': No such file or directory
> sed: can't read /etc/nginx/nginx.conf: No such file or directory
> sed: can't read /usr/local/hestia/nginx/conf/nginx.conf: No such file or directory
> Failed to start nginx.service: Unit nginx.service not found.
> Error: nginx start failed

rssh error also showed up in the beggining but I manually installed it before copy paste here.

**Comment by ScIT-Raphael** [comment]
please add a new issue and also inform about all steps you did.

**Comment by HueyGeek** [comment]
Hi !

I also got this error with Ubuntu 16.04.

The bug is because setpriv is declared as part of the packages to install (end of line #43), only that this command is not a package by itself, but is part of util-linux.

I corrected by changing the declaration to util-linux, instead of setpriv.

**Comment by ScIT-Raphael** [comment]
Please open a new issue and follow the issue template to provide all needed informations. Setpriv was set for 1.1.0 which is currently not released, probably you use the wrong installer.

**Comment by ScIT-Raphael** [comment]
And probaly its a good idea to switch to ubuntu 18.04 :)

**Comment by HueyGeek** [comment]
No, I really did it on purpose, to be able to enjoy the latest features, and for some reason, the VPS provider does not have ubuntu 18.04 available

Thanks for the suggestion.
